functional-tests/fn_bridge_withdraw_happy.py

```diff
@@ -0,0 +1,239 @@
+        ctx.set_env(BasicEnvConfig(pre_generate_blocks=101))
+        self.logger = get_logger("BridgeWithdrawHappyTest")
+
+    # TODO: This needs refactoring to be more clear
+    def main(self, ctx: flexitest.RunContext):
+        address = get_address(0)
+        withdraw_address = get_address(1)
+        self.logger.debug(f"Address: {address}")
+        self.logger.debug(f"Change Address: {withdraw_address}")
+        self.logger.debug(f"Gas: {WITHDRAWAL_GAS_FEE}")
+
+        btc = ctx.get_service("bitcoin")
+        seq = ctx.get_service("sequencer")
+        reth = ctx.get_service("reth")
+
+        seqrpc = seq.create_rpc()
+        btcrpc: BitcoindClient = btc.create_rpc()
+        rethrpc = reth.create_rpc()
+
+        btc_url = btcrpc.base_url
+        btc_user = btc.props["rpc_user"]
+        btc_password = btc.props["rpc_password"]
+
+        self.logger.debug(f"BTC URL: {btc_url}")
+        self.logger.debug(f"BTC user: {btc_user}")
+        self.logger.debug(f"BTC password: {btc_password}")
+
+        web3: Web3 = reth.create_web3()
+        web3.eth.default_account = web3.address
+        el_address = web3.address
+        self.logger.debug(f"EL address: {el_address}")
+
+        # Gas price
+        gas_price = web3.to_wei(1, "gwei")
+        self.logger.debug(f"Gas price: {gas_price}")
+
+        # FIXME: Somehow web3 default account has some balance.
+        # Hence we need to set it to burn it.
+        balance_to_burn = int(rethrpc.eth_getBalance(el_address), 16)
+        self.logger.debug(f"Strata balance to burn: {balance_to_burn}")
+        estimated_gas = web3.eth.estimate_gas(
+            {
+                "from": el_address,
+                "to": BURNER_ADDRESS,
+                "value": balance_to_burn,
+            }
+        )
+        web3.eth.send_transaction(
+            {
+                "from": el_address,
+                "to": BURNER_ADDRESS,
+                "value": balance_to_burn - (estimated_gas * gas_price),
+                "gas": estimated_gas,
+                "gasPrice": gas_price,
+            }
+        )
+        wait_until(lambda: int(rethrpc.eth_getBalance(el_address), 16) == 0)
+        # Ok now we have a clean state
+        self.logger.debug("Strata balance is zero")
+
+        # Get operators pubkey and musig2 aggregates it
+        bridge_pk = get_bridge_pubkey(seqrpc)
+        self.logger.debug(f"Bridge pubkey: {bridge_pk}")
+
+        seq_addr = seq.get_prop("address")
+        self.logger.debug(f"Sequencer Address: {seq_addr}")
+        bridge_pk = get_bridge_pubkey(seqrpc)
+
+        # Generate plenty of BTC to address
+        btcrpc.proxy.generatetoaddress(102, address)
+
+        # Deposit to the EL address
+        # NOTE: we need 2 deposits to make sure we have funds for gas
+        self.make_drt(ctx, el_address, bridge_pk)
+        self.make_drt(ctx, el_address, bridge_pk)
+        wait_until(lambda: int(rethrpc.eth_getBalance(el_address), 16) > 0)
```

Comment:
I think just having a `> 0` might not account for both DRT to be processed. Maybe we can be certain with two wait_until checks. For practical purpose, this should probably be fine though.

Reply:
Done in a663e7424a18213078973394beca7685ab05d45f.